source/grid/grid_out.cc

```diff
@@ -2590,9 +2590,55 @@ void GridOut::write_mesh_per_processor_as_vtu (const Triangulation<dim,spacedim>
       patches.push_back(patch);
     }
 
-  const std::string new_file = (filename_without_extension + ".proc" +
-                                Utilities::int_to_string (tria.locally_owned_subdomain(), 4) +
-                                ".vtu");
+  // only create .pvtu file if running in parallel
+  // if not, just create a .vtu file with no reference
+  // to the processor number
+  std::string new_file = filename_without_extension + ".vtu";
+  if (const parallel::Triangulation<dim,spacedim> *tr =
+        dynamic_cast<const parallel::Triangulation<dim,spacedim> *>(&tria))
+    {
+      new_file = filename_without_extension + ".proc" +
+                 Utilities::int_to_string (tr->locally_owned_subdomain(), 4) +
+                 ".vtu";
+
```

Comment:
@drwells Is this a good fix for the `new_file` variable? I initialize the string before the if-statement as if we do not have a `parallel::Triangulation`, then change it if we do. This allows me to get rid of the entire else-statement below.

Reply:
Yup :+1: